pkg/api/api.go

```diff
@@ -33,6 +34,15 @@ type APIHandler struct {
 	Expire      time.Time
 	Delay       time.Duration
 	metrics     *metrics
+	history     []RequestRecord
```

Comment:
I really think we should make this a bounded resource.  For example I just sent 1/2 billion requests to mock-es but I had no need for the request body.  In that instance I'd like to be able to avoid the memory, cpu and locking costs of storing the body.  What if we added a flag to set the number of recorded bodies, 0 would turn the feature off.

Reply:
I did a quick implementation at https://github.com/leehinman/mock-es/tree/save-request-body , far from perfect but hopefully is gives you a better idea of what I was thinking of

Reply:
Thanks! I didn’t mean to be rude by not replying. I saw the messages but didn’t have the time to prioritize this. I really appreciate you going out of your way to showcase the idea. I’ll get back to this soon and finish up the work.